tensorflow/python/keras/engine/training.py

```diff
@@ -1347,19 +1369,24 @@ def evaluate(self,
       logs = {}
       self.test_function = self.make_test_function()
       self._test_counter.assign(0)
+      epoch_test_function = self._get_compiled_epoch_function(
```

Comment:
@omalleyt12 Do you think it would be useful to cache this function similar to what's done with `self._eval_data_handler`?

Reply:
@omalleyt12 Can you please take a look on above comments from @lgeiger. Thanks!